patch/atmega328pb.yaml

```diff
@@ -0,0 +1,41 @@
+
+# Remove index suffix from all registers and fields
+SPI0:
+  _strip_end:
+    - "0"
+SPI1:
+  _strip_end:
+    - "1"
+  SPSR:
+    _strip_end:
+      - "1"
+  SPCR:
+    _strip_end:
+      - "1"
+
+# Remove index suffix from all registers and fields
+TWI0:
+  _strip_end:
+    - "0"
+TWI1:
+  _strip_end:
+    - "1"
+  TWCR:
+    _strip_end:
+      - "1"
+  TWSR:
+    _strip_end:
+      - "1"
```

Comment:
Hmm, I'm unsure about this ... This isn't the first time we're fighting with the odd naming scheme of AVR registers but I'm inclined to try sticking with the original names as much as possible.  In my opinion, it's better to have the macros from `avr-hal` flexible enough to take whatever names Atmel gave them than to try patching them all to fit.

The reason is simple: This way we have to deal with it once in the macro instead of everytime a new ATDF file with even more creative postfixes shows up.  Even worse, sometimes the numbers aren't at the end, but embedded somewhere in the middle of a register name.  Additionally, this makes cross-referencing C code with Rust easier as the names stay the same everywhere.

---

I know the HAL currently isn't very consistent about this; `twi` makes the HAL crate spell out all register names entirely while `spi` doesn't expose the names at all currently.  I've recently updated at least the `usart` to a flexible macro that uses [`paste`](https://crates.io/crates/paste) (see [`avr-hal-generic/src/serial.rs`](https://github.com/Rahix/avr-hal/blob/master/avr-hal-generic/src/serial.rs)).  This seems to be working well so at some point `twi` and `spi` should probably follow suit.

Reply:
I'm not sure either. The pros for doing it this way is that you don't have to write "specific but identical" code for the peripheral instances. In this special case the registers are suffixes with index, and the fields are too for the second device, but not the first. So you have to supply register field names as well :-/ and as you said, if the index is in the middle of the name you can't supply a empty suffix to the macro.

The only other devices I have worked with are stm32s, and they are doing the fixups in the "device" crate, so that's the primary reason I did it this way here. But I do think it's nice to fix the naming in some cases as they tend to be inconsistent for no really good reason and it just make reading and using them harder.